### ISSUE TITLE
Status Code:414 Request-URI Too Long

### ISSUE THREAD

**Issue by note89** [jahewson/elm-graphql#22]
Hi awesome library, i wonder if something went amiss when implementing POST for mutations since I'm getting this. 
```
Request URL:https://ahead.life/graphql?query=mutation+createEventMutation(%24mutateParam%3A+CreateEventInput!)+%7B+create_event(input%3A+%24mutateParam)+%7B+event_edge+%7B+node+%7B+id+url+title+creator+%7B+id+name+status+%7B+image_src+%7D+%7D+short_text+full_text+start_unix_timestamp+end_unix_timestamp+images+%7B+original+w128+w256+w512+w64+%7D+video_src+latitude+longitude+location+zoom+radius+has_watered+popularity_value+weighted_value+views_value+comments_count+%7D+%7D+%7D+%7D&operationName=createEventMutation&variables=%7B%22mutateParam%22%3A%7B%22clientMutationId%22%3A%22CreateClientMutationId%22%2C%22creator%22%3A%22dXNlcjplYjBjNGU1Yy1kZTNjLTQ5Y2QtYjIzZi05OTIzMjhjOTY3NzU%3D%22%2C%22end_unix_timestamp%22%3A1471867740%2C%22full_text%22%3Anull%2C%22group_tags%22%3A%5B%22Z3JvdXA6ZDU5ODgyN2ItN2Q5NS00NjQ1LWFmZDMtY2FiZjBjN2NjNDAw%22%2C%22Z3JvdXA6NmQ3YjA3ZDEtYzhhMC00ZDgyLTliNGItNTMzZjUwNjNmMzQw%22%2C%22Z3JvdXA6Yzk2NDlmMmMtNDQ3Ny00OTZmLWIxOTktYjdkNjc4YjcwMmFj%22%2C%22Z3JvdXA6Yjg3MGQ5NTQtODRhYy00YjVlLWFmZTktYTc0NDhhOWYzZTNl%22%2C%22Z3JvdXA6OTkwNTgwNjAtMTczNC00OGNjLThiY2YtZGY4NGNkNmRhYzZj%22%2C%22Z3JvdXA6ZGY2MDhmYjgtMTU3Yi00MDgxLThjNjgtZTA1YTQyOTQ1ZDg3%22%2C%22Z3JvdXA6YmE1ZjlkOWYtMzY0ZS00MzEzLTk1MmItMzBiY2ZkMTkzOWIx%22%5D%2C%22image_src%22%3A%22http%3A%2F%2Fi.imgur.com%2FUID947T.jpg%22%2C%22latitude%22%3A58.40186906886139%2C%22location%22%3A%22Vallav%C3%A4gen+14%2C+582+15+Link%C3%B6ping%2C+Sweden%22%2C%22longitude%22%3A15.595049858093262%2C%22short_text%22%3A%22Lorem+ipsum+dolor+sit+amet%2C+consectetur+adipiscing+elit.+Pellentesque+a+erat+dolor.+Duis+non+turpis+sem.+Maecenas+tempus+venenatis+velit%2C+eu+dapibus+lorem+iaculis+quis.+Duis+vehicula+ornare+ipsum+ut+pulvinar.+Aenean+sagittis+ipsum+quis+libero+viverra%2C+quis+interdum+ipsum+porta.+Etiam+hendrerit+ipsum+sit+amet+ex+dapibus+accumsan.+Integer+vulputate+eu+ligula+ac+luctus.+Phasellus+a+tincidunt+nunc.+Nam+sit+amet+blandit+lectus%2C+in+tristique+urna.+Cras+molestie%2C+nunc+at+rhoncus+suscipit%2C+nisl+sem+scelerisque+quam%2C+in+commodo+erat+ligula+sed+nisl.+Fusce+malesuada+porta+augue.+Aliquam+tristique+elit+augue%2C+eu+pharetra+erat+aliquam+id.+Sed+vel+erat+ac+lectus+accumsan+tempus.+In+quis+condimentum+enim.+Morbi+ut+eros+ultrices%2C+finibus+augue+eget%2C+auctor+magna.%5Cn%5CnInterdum+et+malesuada+fames+ac+ante+ipsum+primis+in+faucibus.+Sed+suscipit+magna+nulla%2C+et+bibendum+lorem+scelerisque+sed.+Donec+sollicitudin+eros+ligula%2C+ac+placerat+sem+tempor+nec.+Cum+sociis+natoque+penatibus+et+magnis+dis+parturient+montes%2C+nascetur+ridiculus+mus.+Nam+vitae+leo+eleifend%2C+faucibus+dui+non%2C+sodales+ante.+Phasellus+quam+felis%2C+maximus+nec+lectus+vitae%2C+feugiat+commodo+nibh.+Ut+vitae+rhoncus+est%2C+eu+pulvinar+mauris.+Donec+egestas+nisi+arcu%2C+vitae+tincidunt+libero+eleifend+at.+Donec+vulputate+eros+vitae+ex+ornare+accumsan.+Aliquam+sit+amet+felis+eu+ipsum+gravida+eleifend+quis+mattis+purus.+Ut+ac+tincidunt+odio%2C+quis+ultrices+orci.+Praesent+vehicula+purus+non+sapien+tincidunt+interdum.+Maecenas+consequat+imperdiet+purus+in+mollis.+Etiam+commodo+ut+justo+eget+aliquam.+Praesent+eget+risus+enim.%5Cn%5CnMauris+ac+porttitor+sem.+Curabitur+id+lacus+ultricies%2C+tempor+tellus+ut%2C+egestas+augue.+Suspendisse+finibus+justo+sit+amet+tempor+tincidunt.+Ut+cursus+est+neque%2C+in+porttitor+erat+vehicula+quis.+Mauris+et+nunc+egestas%2C+sodales+felis+vitae%2C+sodales+erat.+Quisque+orci+purus%2C+efficitur+sit+amet+molestie+ac%2C+consequat+id+odio.+Etiam+ac+justo+tristique%2C+lacinia+velit+eget%2C+blandit+libero.+Nunc+tempor+ligula+in+tincidunt+rutrum.+Nullam+dapibus+pulvinar+commodo.+In+tincidunt+blandit+rutrum.+Morbi+feugiat+neque+et+tincidunt+tempor.+Integer+sem+nisi%2C+posuere+efficitur+ante+eu%2C+commodo+luctus+velit.+Vestibulum+tristique+leo+vel+velit+aliquam+semper.+Nam+elit+est%2C+pulvinar+eget+lacus+non%2C+vestibulum+pharetra+sem.+Vestibulum+rutrum%2C+tellus+a+fermentum+bibendum%2C+magna+nunc+laoreet+lorem%2C+id+fermentum+risus+nisl+a+neque.%5Cn%5CnQuisque+lobortis+bibendum+dolor+efficitur+vehicula.+Nunc+non+velit+aliquet%2C+lacinia+est+nec%2C+imperdiet+dui.+Sed+ultricies+enim+accumsan+enim+tristique+varius.+Nunc+sit+amet+purus+convallis%2C+tincidunt+urna+in%2C+sodales+lorem.+Etiam+sit+amet+leo+lectus.+Etiam+placerat+vestibulum+erat.+Donec+mollis+nunc+egestas+ex+aliquam%2C+a+eleifend+velit+mattis.+Proin+tortor+ex%2C+rutrum+eget+lobortis+molestie%2C+imperdiet+non+arcu.+Nunc+nec+egestas+neque%2C+at+placerat+ligula.+Vestibulum+bibendum+leo+a+purus+volutpat%2C+vitae+placerat+lorem+semper.+Pellentesque+mauris+eros%2C+rhoncus+ac+condimentum+id%2C+rutrum+et+mi.+Etiam+eu+orci+eget+nulla+convallis+viverra.+Phasellus+non+ipsum+facilisis%2C+vestibulum+felis+eget%2C+molestie+mauris.+Aliquam+erat+sem%2C+vulputate+id+nibh+in%2C+ornare+dictum+justo.+Donec+eu+orci+ut+lorem+tempor+tempor+eu+id+urna.%5Cn%5CnPraesent+quis+dolor+vel+ex+vestibulum+egestas.+Mauris+interdum%2C+purus+sed+bibendum+varius%2C+massa+sapien+maximus+urna%2C+in+consectetur+purus+nisl+a+mi.+Vivamus+in+neque+id+lorem+fringilla+rutrum.+Duis+vel+urna+enim.+Fusce+ornare+metus+enim%2C+sed+vehicula+nulla+ullamcorper+at.+Fusce+sagittis+congue+rutrum.+Morbi+ac+tincidunt+mauris%2C+in+sagittis+dolor.+Cum+sociis+natoque+penatibus+et+magnis+dis+parturient+montes%2C+nascetur+ridiculus+mus.+Nunc+finibus+placerat+porttitor.+Nullam+eget+facilisis+lectus%2C+ut+interdum+sem.+Phasellus+commodo+arcu+nec+risus+rhoncus+semper.+Vivamus+euismod+in+nisi+non+porttitor.+Integer+sit+amet+metus+ut+sapien+feugiat+fermentum.+Nam+vel+quam+non+urna+eleifend+semper+luctus+vitae+velit.+Pellentesque+vel+leo+gravida%2C+sollicitudin+sapien+in%2C+vestibulum+tortor.%22%2C%22start_unix_timestamp%22%3A1471694940%2C%22title%22%3A%22VALLA+RAVE+2016!%22%2C%22url%22%3A%22https%3A%2F%2Fwww.facebook.com%2Fevents%2F1408620579445386%2F%22%2C%22video_src%22%3Anull%2C%22zoom%22%3A16%7D%7D
Request Method:POST
Status Code:414 Request-URI Too Long
Remote Address:178.62.249.126:443
```

https://httpstatuses.com/414

**Comment by note89** [comment]
here is my solution, but i saw that there was a PR aswell 
```
mutation : String -> String -> String -> String -> Decoder a -> Task Http.Error a
mutation url query operation variables decoder =
    let
        body =
            JE.object
                [ ( "query", JE.string query )
                , ( "operationName", JE.string operation )
                , ( "variables", JE.string variables )
                ]

        request =
            { verb = "POST"
            , headers =
                [ ( "Accept", "application/json" )
                , ( "Content-Type", "application/json" )
                ]
            , url = url
            , body = Http.string <| JE.encode 4 body
            }
    in
        Http.fromJson (queryResult decoder) (Http.send Http.defaultSettings request)
```

**Comment by proProbe** [comment]
+1

**Comment by jahewson** [comment]
This should be fixed by 3d6aacc91091fd53bf7361c514ff17eb5f8e6c1f, which moves the payload into the body.
